### PR TITLE
Updated BasicHistoryApi example for Kotlin

### PR DESCRIPTION
**Pull request by garanj**
- Migrated to Kotlin
- Added a sign out function
- Updated dependencies and migrated to AndroidX
- Migrated to more recent GoogleSignIn methods
- Added check for OAuth permissions before each action, e.g. before "Update", "Delete" from the menu, to handle case where permissions have been revoked.

### CODE REVIEW

File: BasicHistoryApi/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
-Copyright 2014 Google, Inc.
+  Copyright 2019 Google LLC
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/AndroidManifest.xml]: You actually don't need to update the year for existing files (leave them alone generally).

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+import com.google.android.gms.fitness.request.DataReadRequest
+import com.google.android.gms.fitness.request.DataUpdateRequest
+import com.google.android.gms.fitness.result.DataReadResponse
+import com.google.android.gms.tasks.Task
+import java.text.DateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+const val TAG = "BasicHistoryApi"
+
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: For the top level class name, make sure you always include a full comment on what the class does. Make sure it is descriptive, short, and gives a good overview for the developer to get started.

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+import com.google.android.gms.fitness.Fitness
+import com.google.android.gms.fitness.FitnessOptions
+import com.google.android.gms.fitness.data.*
+import com.google.android.gms.fitness.request.DataDeleteRequest
+import com.google.android.gms.fitness.request.DataReadRequest
+import com.google.android.gms.fitness.request.DataUpdateRequest
+import com.google.android.gms.fitness.result.DataReadResponse
+import com.google.android.gms.tasks.Task
+import java.text.DateFormat
+import java.util.*
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: Our style guidelines prohibit using .* imports.

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+import androidx.core.widget.TextViewCompat
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.fit.samples.common.logger.Log
+import com.google.android.gms.fit.samples.common.logger.LogView
+import com.google.android.gms.fit.samples.common.logger.LogWrapper
+import com.google.android.gms.fit.samples.common.logger.MessageOnlyLogFilter
+import com.google.android.gms.fitness.Fitness
+import com.google.android.gms.fitness.FitnessOptions
+import com.google.android.gms.fitness.data.*
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: Our style guidelines prohibit using .* imports. It's confusing for developers trying to figure out what is and isn't included.

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+            requestCode?.let {
+                GoogleSignIn.requestPermissions(
+                        this,
+                        requestCode,
+                        getGoogleAccount(), fitnessOptions)
+            }
+        } else signInAction()
+    }
+
+    /**
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: I would remove this comment. The function name is descriptive enough. :)

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+                .addOnSuccessListener {
+                    Log.i(TAG, "Successfully deleted today's step count data.")
+                }
+                .addOnFailureListener { e ->
+                    Log.e(TAG, "Failed to delete today's step count data.", e)
+                }
+    }
+
+    /**
+     * Updates and reads data by chaning [Task] from [.updateData] and [ ][.readHistoryData].
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: chaining?

File: BasicHistoryApi/app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,11 +1,11 @@
 <!--
-  Copyright 2014 Google, Inc.
+  Copyright 2019 Google LLC
```
Comment (codingjeremy) [inline BasicHistoryApi/app/src/main/res/layout/activity_main.xml]: Again, you don't have to do this. You can leave changes this time if you want, I just wouldn't worry about it moving forward.

**Comment by garanj** [comment]
I'll check that video out - sounds a nice idea! And will take on the renaming in another PR as your suggest.

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt (hunk 4 above)
Comment (garanj) [reply BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: Done, thanks!

File: BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt (hunk 2 above)
Comment (garanj) [reply BasicHistoryApi/app/src/main/java/com/google/android/gms/fit/samples/basichistoryapi/MainActivity.kt]: Apologies! I've transferred over the one from the Java version. Don't know how it went missing!

**Pull request merged**